src/services/user.service.js

```diff
@@ -0,0 +1,46 @@
+let users = [];
+let nextUser = 1;
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((user) => user.id === id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    id: nextUser++,
+    name,
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(id);
```

Comment:
Consider adding a check to handle cases where the user with the given ID does not exist before attempting to update it. This will prevent potential errors when trying to update a non-existent user.